abipy/abio/factories.py

```diff
@@ -416,7 +418,103 @@ def g0w0_with_ppmodel_inputs(structure, pseudos,
+        sigma_nband: Number of bands used to compute the self-energy (default is nscf_nband)
+        workdir: Working directory.
+        manager: :class:`TaskManager` instance.
+        nksamll: if not None, a DFT bandstucture calculation will be added after the sc run
+        extra_abivars: Dictionary with extra variables passed to ABINIT.
+    """
+    print(scf_nband)
+    if gamma:
+        if kppa == 1:
+            scf_ksampling = KSampling.gamma_centered(kpts=(1, 1, 1))
+            nscf_ksampling = KSampling.gamma_centered(kpts=(1, 1, 1))
+        elif kppa == 2:
+            scf_ksampling = KSampling.gamma_centered(kpts=(2, 2, 2))
+            nscf_ksampling = KSampling.gamma_centered(kpts=(2, 2, 2))
+        elif kppa < 0:
+            scf_ksampling = KSampling.gamma_centered(kpts=(-kppa, -kppa, -kppa))
+            nscf_ksampling = KSampling.gamma_centered(kpts=(2, 2, 2))
+        elif kppa <= 13: 
+            scf_ksampling = KSampling.gamma_centered(kpts=(kppa, kppa, kppa))
+            nscf_ksampling = KSampling.gamma_centered(kpts=(kppa, kppa, kppa))
+        else:
+            scf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0, shifts=(0, 0, 0))
+            nscf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0, shifts=(0, 0, 0))
+    else:
+        #this is the original behaviour before the devellopment of the gwwrapper
+        scf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0)
+        nscf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0)
+
+    scf_electrons = aobj.Electrons(spin_mode=spin_mode, smearing=smearing, algorithm=scf_algorithm,
+                                   charge=charge, nband=scf_nband, fband=None)
+  
+    multis = []
+
+    to_add = {}
+
+    if "istwfk" not in extra_abivars:
+        extra_abivars["istwfk"] = "*1"
+
+    scf_diffs = [] 
+
+    for k in extra_abivars.keys():
+        if k[-2:] == '_s':
+            var = k[:len(k)-2]
+            values = extra_abivars.pop(k)
+            to_add.update({k: values[-1]})
+            for value in values:
+		diff_abivars = {}
+                diff_abivars[var] = value
+                diff_abivars['pawecutdg'] = diff_abivars['ecut']*2
+                scf_diffs.append(diff_abivars)
+#                multi_scf.append(ScfStrategy(structure, pseudos, scf_ksampling, accuracy=accuracy,
+#                                                spin_mode=spin_mode, smearing=smearing, charge=charge,
+#                                                scf_algorithm=None, nband=scf_nband, **extra_abivars))
+#
+#    print(scf_diffs)
+#    print(extra_abivars)
+#    print(to_add)
+
+    print(scf_diffs)
+
+    multi_scf = MultiDataset(structure, pseudos, ndtset=len(scf_diffs))
+   
+    multi_scf.set_vars(scf_ksampling.to_abivars())
+    multi_scf.set_vars(scf_electrons.to_abivars())
+    multi_scf.set_vars(extra_abivars)
+    for vars, abinput in zip(scf_diffs, multi_scf):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'vars'](https://www.codacy.com/app/gmatteo/abipy/file/2514241404/issues/source?bid=3176346&fileBranchId=3277671#l508)